### PR TITLE
add documentation on howto build just rustc without libstd to the build system

### PR DESCRIPTION
**Pull request by flo-l**
I searched for days until I found this, maybe this helps other poor souls :joy:

for reference #33990

### CODE REVIEW

**Comment by flo-l** [comment]
cc @alexcrichton

**Comment by alexcrichton** [comment]
We may want to replace the "x86_64-unknown-linux-gnu" here with "$host" perhaps as it'll differ depending on where you're compiling. The explanation may also want to be accompanied with a warning indicating that this may not work as the standard library really does need to be recompiled to be guaranteed to work.

**Comment by flo-l** [comment]
good catch with x86_64-unknown-linux-gnu, fixed it and added some more words of warning regarding libstd.

File: Makefile.in [hunk 1]
```diff
@@ -62,6 +62,7 @@
 #   * tidy - Basic style check, show highest rustc error code and
 #     the status of language and lib features
 #   * rustc-stage$(stage) - Only build up to a specific stage
+#   * x86_64-unknown-linux-gnu/stage1/bin/rustc - Only build stage1 rustc, not libstd.
```
Comment (alexcrichton) [inline Makefile.in]: One more "$host" here

**Comment by flo-l** [comment]
fixed again!

**Pull request merged**